flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImpl.java

```diff
@@ -102,11 +102,11 @@ public void start(long checkpointId, CheckpointOptions checkpointOptions) {
 		LOG.debug("{} starting checkpoint {} ({})", taskName, checkpointId, checkpointOptions);
 		ChannelStateWriteResult result = new ChannelStateWriteResult();
 		ChannelStateWriteResult put = results.computeIfAbsent(checkpointId, id -> {
-			Preconditions.checkState(results.size() < maxCheckpoints, "results.size() > maxCheckpoints", results.size(), maxCheckpoints);
+			Preconditions.checkState(results.size() < maxCheckpoints, String.format("can't start %d, results.size() > maxCheckpoints: %d, %d, %s", checkpointId, results.size(), maxCheckpoints, taskName));
```

Comment:
nit: usually we add task name to front, also it would be nice to use `%d > %d` for the comparison values.